src/Microsoft.AspNetCore.SignalR.Common/Protocol/HandshakeResponseMessage.cs

```diff
@@ -9,23 +9,42 @@ namespace Microsoft.AspNetCore.SignalR.Protocol
     public class HandshakeResponseMessage : HubMessage
     {
         /// <summary>
+        /// Depreciated.
         /// An empty response message with no error.
         /// </summary>
-        public static readonly HandshakeResponseMessage Empty = new HandshakeResponseMessage(null);
+        public static readonly HandshakeResponseMessage Default = new HandshakeResponseMessage(error: null);
 
         /// <summary>
         /// Gets the optional error message.
         /// </summary>
         public string Error { get; }
 
+        /// <summary>
+        /// Highest minor protocol version that the server supports.
+        /// </summary>
+        public int MinorVersion { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HandshakeResponseMessage"/> class.
+        /// If you are intending to send back an error, the minor version doesn't matter.
+        /// </summary>
+        /// <param name="error"></param>
+        public HandshakeResponseMessage(string error) : this(null, error) { }
+
+        public HandshakeResponseMessage(int minorVersion) : this(minorVersion, null) { }
+
+        public HandshakeResponseMessage(IHubProtocol protocol) 
```

Comment:
This ctor feels a little weird. IMO the response message shouldn't know about the IHubProcotol.

Reply:
I agree that it's weird, but I felt bad writing the same "try to cast, default to 0" logic in two different places. Is there a better place (than a ctor) for holding this logic?

Reply:
If you need to do that logic in multiple places, you could write an extension method:

```csharp
public static class HubProtocolMinorVersionExtensions
{
    public static int GetMinorVersion(this IHubProtocol protocol)
    {
        // Do the magic.
    }
}
```

Then it could be called from somewhere else just like a regular method: `protocol.GetMinorVersion()`.